src/basic/fd-util.h

```diff
@@ -104,7 +104,15 @@ static inline int make_null_stdio(void) {
                 0;                              \
         })
 
-
 int fd_reopen(int fd, int flags);
 int read_nr_open(void);
 int btrfs_defrag_fd(int fd);
+
+/* The maximum length a buffer for a /proc/self/fd/<fd> path needs */
+#define PROC_FD_PATH_MAX \
+        (STRLEN("/proc/self/fd/") + DECIMAL_STR_MAX(int) + 1U)
```

Comment:
Why `+1U` here?

Reply:
Yeah, it's unnecessary.

Reply:
as space for the final NUL byte.

now DECIMAL_STR_MAX() includes one too, but we had a discussion a few years back where we came to the conclusion that that was a mistake, and we should eventually drop the extra +1 from DECIMAL_STR_MAX(), since it's typically uses as part of a calculation, not as fully coverage. hence i started to always add the +1 already because it's safer that way then the other way.

Reply:
No, please don't do that. If we ever decide to make `DECIMAL_STR_MAX` not include the extra byte, we'll automatically replace all calls to `DECIMAL_STR_MAX()` with `DECIMAL_STR_MAX()+1`. There is simply too many of them to do inspect and adjust all sites by hand. And this means that adding +1 *right now* is counterproductive — we'll end up with +2 unnecessarily.